generics/services/kendra-service.js

```diff
@@ -7,56 +7,120 @@
 
 //dependencies
 
-let urlPrefix = 
-process.env.KENDRA_SERIVCE_HOST + 
-process.env.KENDRA_SERIVCE_BASE_URL +
-process.env.URL_PREFIX; 
+const apiBaseUrl =
+    process.env.KENDRA_SERIVCE_HOST +
+    process.env.KENDRA_SERIVCE_BASE_URL +
+    process.env.URL_PREFIX;
 
 const request = require('request');
+const fs = require('fs');
 
 /**
-  * Get platform user roles
-  * @function
-  * @name platformUserProfile
-  * @returns {Promise} returns a promise.
-*/
+ * To upload file to cloud 
+ * @name uploadFile
+ * @param {String} filePath filePath of the file to upload
+ * @param {String} uploadPath - location of bucket where to upload
+ * @returns {json} - upload file details
+ */
+function uploadFile(filePath, uploadPath) {
+    return new Promise(async (resolve, reject) => {
+        try {
+            let options = {
+                "headers": {
+                    'Content-Type': "application/json",
+                    "internal-access-token": process.env.INTERNAL_ACCESS_TOKEN
+                },
+                formData: {
+                    filePath: uploadPath,
+                    bucketName: process.env.STORAGE_BUCKET,
+                    file: fs.createReadStream(filePath)
 
-var getPreSignedUrl = function ( token,body,endpoint ) {
+                }
+            };
 
-    const getPresignedUrl = 
-    urlPrefix + endpoint;
-    
-    return new Promise(async (resolve, reject) => {
+             let endpoint = "";
+
+            let cloudStorage = process.env.CLOUD_STORAGE;
+            if (cloudStorage == constants.common.AWS_SERVICE) {
+                endpoint = constants.endpoints.UPLOAD_TO_AWS;
+            } else if (cloudStorage == constants.common.GOOGLE_CLOUD_SERVICE) {
+                endpoint = constants.endpoints.UPLOAD_TO_GCP;
+            } else if (cloudStorage == constants.common.AZURE_SERVICE) {
+                endpoint = constants.endpoints.UPLOAD_TO_AZURE;
+            }
+
+            let apiUrl =
+                apiBaseUrl + endpoint;
+
+            request.post(apiUrl, options, callback);
+            function callback(err, data) {
+
+                if (err) {
+                    return reject({
+                        message: constants.apiResponses.KENDRA_SERVICE_DOWN
+                    });
+                } else {
+                    return resolve(JSON.parse(data.body));
+                }
+            }
+        } catch (error) {
+            reject(error);
+        }
+    })
+}
+
+/**
+ * To downloadable Urls for cloud files 
```

Comment:
@rakeshSgr - Description is. - "To get downloadable Urls of files in cloud storage"